crates/autopilot/src/infra/blockchain/contracts.rs

```diff
@@ -86,10 +105,51 @@ impl Contracts {
     pub fn authenticator(&self) -> &contracts::GPv2AllowListAuthentication {
         &self.authenticator
     }
+
+    /// Removes solver from the allow-list in the authenticator contract. This
+    /// solver will no longer be able to settle.
+    pub async fn remove_solver(
+        &self,
+        solver: domain::eth::Address,
+    ) -> Result<Option<TransactionResult>, Error> {
+        let Some(authenticator_eoa) = self.authenticator_eoa.clone() else {
+            return Ok(None);
+        };
```

Comment:
Seems like it would be more appropriate to pass in a non-optional EOA. Then a `CircuitBreaker` struct could be the only owner of this EOA since it don't foresee this to be very relevant to the other autopilot code.